src/test/fuzz/cluster_linearize.cpp

```diff
@@ -446,19 +446,36 @@ FUZZ_TARGET(clusterlin_components)
     // Construct a depgraph.
     SpanReader reader(buffer);
     DepGraph<TestBitSet> depgraph;
+    std::vector<DepGraphIndex> linearization;
     try {
         reader >> Using<DepGraphFormatter>(depgraph);
     } catch (const std::ios_base::failure&) {}
 
     TestBitSet todo = depgraph.Positions();
     while (todo.Any()) {
-        // Find a connected component inside todo.
-        auto component = depgraph.FindConnectedComponent(todo);
+        // Pick a transaction in todo, or nothing.
+        std::optional<DepGraphIndex> picked;
+        {
+            uint64_t picked_num{0};
+            try {
+                reader >> VARINT(picked_num);
+            } catch (const std::ios_base::failure&) {}
+            if (picked_num < todo.Size() && todo[picked_num]) {
```

Comment:
I modified it to not gate on existence in `todo`, and nothing seems to be breaking. Shouldn't this cause an `Assume` failure when holes existed in the serialization and the chosen number selects that unused hole?

Reply:
99.9% chance I was running the wrong binary, it instantly crashes for me this time, feel free to resolve